dockstore-webservice/src/main/java/io/dockstore/webservice/languages/CWLHandler.java

```diff
@@ -391,65 +415,51 @@ public Optional<String> getContent(String mainDescriptorPath, String mainDescrip
+        workflow.put("id", "_dockstore_wrapper");
         workflow.put("class", "Workflow");
         workflow.put("inputs", Map.of());
         workflow.put("outputs", Map.of());
         workflow.put("steps", Map.of("tool", Map.of("run", tool, "in", List.of(), "out", List.of())));
         return workflow;
     }
 
-    @SuppressWarnings("checkstyle:ParameterNumber")
-    private void processWorkflow(Workflow workflow, Map<String, Map> parentRequirements, Map<String, Map> parentHints, int depth, String parentStepId, LanguageHandlerInterface.Type type, Preprocessor preprocessor, ToolDAO dao, List<Pair<String, String>> nodePairs, Map<String, ToolInfo> toolInfoMap, Map<String, String> stepToType, Map<String, DockerInfo> nodeDockerInfo) {
-        Yaml yaml = new Yaml();
-        Gson gson = CWL.getTypeSafeCWLToolDocument();
-
-        // Join parent and current requirements and hints.
-        Map<String, Map> requirements = joinRequirementsOrHints(parentRequirements, workflow.getRequirements());
-        Map<String, Map> hints = joinRequirementsOrHints(parentHints, workflow.getHints());
-
-        // Store workflow steps in json and then read it into map <String, WorkflowStep>
-        Object steps = workflow.getSteps();
-        String stepJson = gson.toJson(steps);
-        Map<String, WorkflowStep> workflowStepMap;
-        if (steps instanceof ArrayList) {
-            ArrayList<WorkflowStep> workflowStepList = gson.fromJson(stepJson, new TypeToken<ArrayList<WorkflowStep>>() {
-            }.getType());
-            workflowStepMap = new LinkedTreeMap<>();
-            workflowStepList.forEach(workflowStep -> workflowStepMap.put(workflowStep.getId().toString(), workflowStep));
-        } else {
-            workflowStepMap = gson.fromJson(stepJson, new TypeToken<Map<String, WorkflowStep>>() {
-            }.getType());
-        }
+    private String className(Object obj) {
+        return obj != null ? obj.getClass().getName() : "null object";
+    }
 
-        if (stepJson == null) {
-            LOG.error("Could not find any steps for the workflow.");
-            return;
-        }
+    /**
+     * This function converts the workflow step ID that cwljava returns, which includes the enclosing workflow IDS,
+     * to form that we use internally.
+     * For example, given the id "/#W1/S1/W2/S2", where W1 and W2 are the parent and child workflow IDs and
+     * S1 and S2 are the corresponding workflow step IDs, this function will return "S1.S2".
+     */
+    private String convertStepId(String cwljavaStepId) {
+        List<String> parts = Arrays.asList(cwljavaStepId.replaceFirst("^/", "").split("/"));
+        return NODE_PREFIX + IntStream.range(0, parts.size()).filter(i -> i % 2 == 1).mapToObj(parts::get).collect(Collectors.joining("."));
+    }
 
-        if (workflowStepMap == null) {
-            LOG.error("Error deserializing workflow steps");
-            return;
-        }
+    @SuppressWarnings("checkstyle:ParameterNumber")
```

Comment:
Hah, this feels like it could use a second pass. This is an incredible number of parameters. That said, can be follow-up that requires some thought on how to refactor

Reply:
I removed the `preprocessor` argument, it was no longer needed.  Agreed that the remaining number of parameters is high.   Might make sense to improve this if/when we ever add full DAG support for CWLs, but I feel like right now, there's bigger fish to fry.